### ISSUE TITLE
Trouble running example in Windows 10 environment

### ISSUE THREAD

**Issue by marklnichols** [zmactep/hasbolt-sample-app#2]
Hi -

I'm having trouble getting the example to run correctly in my Windows 10 environment.

First, I have had to set:
    set GRAPHENEDB_BOLT_USER=neo4j
    set GRAPHENEDB_BOLT_PASSWORD=neo4j
    set GRAPHENEDB_BOLT_URL=http://localhost
otherwise I get the error: user error (Authentification failed)
After setting those variables and re-running, the page comes up but in the console I see:

Setting phasers to stun... (port 8080) (ctrl-c to quit)
GET /
  Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
  Status: 200 OK 0s
Prelude.read: no parse
Prelude.read: no parse

If I hit the search button, nothing happens, but in the log I see one more:
Prelude.read: no parse

My Neo4J version is 3.1.0

**Comment by zmactep** [comment]
Hi, @marklnichols !

Try to use double double-quotes like:
```
set GRAPHENEDB_BOLT_USER="\"neo4j\""
set GRAPHENEDB_BOLT_PASSWORD="\"neo4j\""
set GRAPHENEDB_BOLT_URL="\"localhost\""
```

**Comment by marklnichols** [comment]
Hi again,

Well, instead of the Prelude.read: no parse, now I get Prelude.last: empty list.  I tried a few other permutations of quotes but I didn't hit on anything that worked.

**Comment by zmactep** [comment]
Yeah, of course. Look at the example code:
```haskell
bolt <- getEnv "GRAPHENEDB_BOLT_URL"
...
let (host, port) = let sp = last (elemIndices ':' bolt)
                   in (read $ take sp bolt, read $ drop (sp+1) bolt)
```

GRAPHENEDB_BOLT_URL should has format like `localhost:7687` ([link](https://devcenter.heroku.com/articles/graphenedb)).

**Comment by marklnichols** [comment]
Now back to the previous error ;-)

I'll have more time to play with this on the weekend.  Now that I know its related to the environment variables, I'm sure I'll get it to work.  Probably the best is to have some defaults of localhost:8080, neo4j, neo4j if the variables are not set.  I'll can do that and push it back to you if you like.

Cheers,

Mark

**Comment by zmactep** [comment]
Sure, I'll be happy. This was a quick and dirty example to show that has bolt works (btw, I strongly recommend you to update driver to the latest version). Let's make it nicer. :)

**Comment by marklnichols** [comment]
I played around with this a bit.  The environment issues are ironed out, but now I'm getting the error "user error (Unsupported server version)" when trying to use the query button in the UI.  I hard-coded a query I knew worked in my neo4j, but that didn't seem to help. The default config sets the "version" to "1", but I don't know if this has anything to do with the error. I have the most recent version of your libraries and downloaded the latest neo4j as well.

**Comment by zmactep** [comment]
Can I see your connection config?

**Comment by marklnichols** [comment]
Hi -- here is the relevant section of main.hs:

defaultConfig :: BoltCfg
defaultConfig = def {user = "neo4j", password = "neo4j", port = 8080}

-- |Run as stack exec hasbolt-sample-app-exe
main :: IO ()
main = run `catchError` failMsg where 
    run = do    
        config <- readConfig `catchError` const (return defaultConfig)
        envPort <- getEnv "PORT" -- --override with PORT variable if it exists
        let finalPort = fromMaybe (port config) $ readMay envPort 
        let finalConfig = config {port = finalPort}
        runServer config
    readConfig = do
        user <- read <$> getEnv "GRAPHENEDB_BOLT_USER"
        pass <- read <$> getEnv "GRAPHENEDB_BOLT_PASSWORD"
        bolt <- getEnv "GRAPHENEDB_BOLT_URL" -- url including port 
        let (host, port) = let sp = last (elemIndices ':' bolt)
                                   in (read $ take sp bolt, read $ drop (sp+1) bolt)
        return def { user = user, password = pass, host = host, port = port }

**Comment by zmactep** [comment]
And is your BOLT port 8080 really? It is often `7687`.

**Comment by marklnichols** [comment]
Yes, of course! ...it is 7687, sigh... and now I see why there were two separate port variables.  Ok, I'll fix this up and sent it to you as soon as I get a chance.

**Comment by marklnichols** [comment]
Hi -- sorry for the delay, but I got busy at work and wanted to spend a little time looking at the next error.

So now I get:
connect: failed (Connection refused (WSAECONNREFUSED))

I printed out the config, here it is:

BoltCfg {magic = 1616949271, version = 1, userAgent = "hasbolt/1.0", maxChunkSize = 65535, 
         socketTimeout = 5, host = "localhost", port = 7687, user = "neo4j", password = "neo4j"}
